src/game.cpp

```diff
@@ -2204,8 +2204,9 @@ void Game::toggleMinimap(bool shift_pressed)
 
 void Game::toggleFog()
 {
-	m_flags.force_fog_off = !m_flags.force_fog_off;
-	if (m_flags.force_fog_off)
+	bool fog_enabled = g_settings->getBool("enable_fog");
+	g_settings->setBool("enable_fog",!fog_enabled);
```

Comment:
Missing space after `,`
`m_cache_enable_fog` is nowhere updated. Did you test this?

Reply:
Yes I have tested, and actually there is a callback to update it